src/sql/workbench/contrib/notebook/test/browser/htmlMarkdownConverter.test.ts

```diff
@@ -153,6 +154,37 @@ suite('HTML Markdown Converter', function (): void {
 		assert.equal(htmlMarkdownConverter.convert(htmlString), `1. Test\n    1. Test2\n2. Test3`, 'Basic ordered item test failed');
 	});
 
+	test('Should keep < > tag', () => {
+		htmlString = '&lt;test&gt';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '<test>', 'Non-HTML tag test failed to escape');
+		htmlString = '&lt;test&gt<span style="background:red">message</span>&lt;test&gt';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '<test><span style="background:red">message</span><test>', 'Non-HTML tag inside span tag test failed to escape');
+		htmlString = '<h1>&lt;test&gt;<h1>';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '# \\<test\\>', 'Non-HTML tag inside H1 tag test failed to escape');
+		htmlString = '<h2>&lt;test&gt;<h2>';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '## \\<test\\>', 'Non-HTML tag inside H2 tag test failed to escape');
+		htmlString = '<h3>&lt;test&gt;<h3>';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '### \\<test\\>', 'Non-HTML tag inside H3 tag test failed to escape');
+		htmlString = '<a href="https://www.microsoft.com/images/msft.png">&lt;msft&gt</a>';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '[\\<msft\\>](https://www.microsoft.com/images/msft.png)', 'Non-HTML tag as link test failed to escape');
+		htmlString = '<strong>&lt;Bold test&gt;</strong>';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '**\\<Bold test\\>**', 'Basic bold non-HTML tag test failed to escape');
+		htmlString = '<em>&lt;Italicize test&gt;</em>';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '_\\<Italicize test\\>_', 'Basic italicize non-HTML tag test failed to escape');
+		htmlString = '<u>&lt;Underline_test&gt;</u> ';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '<u>&lt;Underline_test&gt;</u>', 'Basic underline non-HTML tag test failed to escape');
+		htmlString = '<ul><li>&lt;test&gt;</li></ul>';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '- \\<test\\>', 'Basic unordered list non-HTML tag item test failed to escape');
+		htmlString = '<ol><li>&lt;test&gt;</li></ol>';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '1. \\<test\\>', 'Basic ordered list non-HTML tag item test failed to escape');
+		htmlString = '<mark>&lt;test&gt;</mark>';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '<mark>&lt;test&gt;</mark>', 'Basic highlighting Non-HTML tag test failed to escape');
+		htmlString = '<mark><h1>&lt;test&gt;</h1></mark>';
+		assert.equal(htmlMarkdownConverter.convert(htmlString), '<mark><h1>&lt;test&gt;</h1></mark>', 'Non-HTML tag inside multiple html tags test failed to escape');
+		htmlString = '<p>&lt;style&gt</p>';
```

Comment:
What about `<p>style</p>`?

Reply:
That would just return the text 'style' as expected.

Reply:
I'm saying that the content contains the string `<p>style</p>` - which if I'm understanding the logic correctly would not escape the <'s and thus still have the bug.

Reply:
So below is how it WYSIWYG and Markdown should treat 'style', and '<style>'. The above example you have, `<p>style</p>` would be the just the text. You would have to include either the escape `\\` or `&lt;` and `&gt;` to keep the <'s. Therefore, there should not be a bug.

![image](https://user-images.githubusercontent.com/23587151/98604815-05155180-22aa-11eb-9b21-c2cbe80f2ad4.png)

Reply:
Ok, so what about `\<styletag\>`? Is that also still escaped correctly? 

I'm not even sure I understand why u, mark and style are special in this regard - why are you having to skip escaping the brackets for only those ones?

Reply:
So the specific edge case we were tackling was style, and it seems that this would be an even further edge case that should be fixed outside this PR. 

For your other question, when looking over the WYSIWYG code you can notice that there is a keep setting that we choose to include which allows us to keep certain tags. https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/notebook/browser/htmlMarkdownConverter.ts#L24
You can read about it more here: https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/notebook/browser/htmlMarkdownConverter.ts#L24